models/issues/pull.go

```diff
@@ -797,14 +797,16 @@ func HasEnoughApprovals(ctx context.Context, protectBranch *git_model.ProtectedB
 
 // GetGrantedApprovalsCount returns the number of granted approvals for pr. A granted approval must be authored by a user in an approval whitelist.
 func GetGrantedApprovalsCount(ctx context.Context, protectBranch *git_model.ProtectedBranch, pr *PullRequest) int64 {
-	sess := db.GetEngine(ctx).Where("issue_id = ?", pr.IssueID).
-		And("type = ?", ReviewTypeApprove).
-		And("official = ?", true).
-		And("dismissed = ?", false)
+	opt := &GetReviewOption{
```

Comment:
How about reuse `db.Find` here?

Reply:
Yes ... in a dedicated refactor pull that dont have to be backported. i'll have to mobe a lot more code around.

If desired i instead can seperate the bugfixes and tjen start refactoring here

Reply:
You can use the latest method on this PR and fall back when backporting.

Reply:
well in this case I split out to get the bugfix in asap and then make the code clean here :)

Reply:
-  #28551
- #28552

Reply:
well all split out pulls are merged now ... moving forward :)